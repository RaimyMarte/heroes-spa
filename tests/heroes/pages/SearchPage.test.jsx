import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes/pages/SearchPage"

describe('pruebas en SearchPage', () => {
    test('debe de mostrarse correctamente con valores por defecto', () => {

        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot()
    })

    test('debe de mostrarse a Batman y el ionput con el con valor del queryString', () => {
        const searchText = 'batman'
        render(
            <MemoryRouter initialEntries={[`/search?q=${searchText}`]}>
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox')
        expect(input.value).toBe(searchText)

        const img = screen.getByRole('img')
        expect(img.src).toContain('/heroes/dc-batman.jpg')

        const primaryAlert = screen.getByLabelText('primaryAlert')
        expect(primaryAlert.style.display).toBe('none')

        const dangerAlert = screen.getByLabelText('dangerAlert')
        expect(dangerAlert.style.display).toBe('none')
    })

    test('debe de mostrar un error si no se encuentra el hero (batman123)', () => {
        const searchText = 'batman123'
        render(
            <MemoryRouter initialEntries={[`/search?q=${searchText}`]}>
                <SearchPage />
            </MemoryRouter>
        )

        const dangerAlert = screen.getByLabelText('dangerAlert')

         expect(dangerAlert.style.display).toBe('')

    })

    test('debe de llamar el navigate a la pantalla nueva', () => {
        const searchText = 'superman'
        render(
            <MemoryRouter initialEntries={[`/search?q=${searchText}`]}>
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox')
        fireEvent.click(input)

    })
})