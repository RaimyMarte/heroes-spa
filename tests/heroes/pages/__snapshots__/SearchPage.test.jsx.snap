// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en SearchPage debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row gy-4"
  >
    <div
      class="col-12 col-md-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-12 col-md-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="primaryAlert"
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="dangerAlert"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
